File: java/test/Ice/background/BackgroundI.java
// **********************************************************************
//
// Copyright (c) 2003-2009 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

import Test.*;

class BackgroundI extends _BackgroundDisp
{
    public void
    op(Ice.Current current)
    {
        _controller.checkCallPause(current);
    }

    public void
    opWithPayload(byte[] seq, Ice.Current current)
    {
        _controller.checkCallPause(current);
    }

    public void
    shutdown(Ice.Current current)
    {
        current.adapter.getCommunicator().shutdown();
    }

    BackgroundI(BackgroundControllerI controller)
    {
        _controller = controller;
    }

    private final BackgroundControllerI _controller;
}

File: csharp/test/Ice/background/BackgroundI.cs
// Copyright (c) ZeroC, Inc.

using Test;

internal class BackgroundI : BackgroundDisp_
{
    public override void op(Ice.Current current)
    {
        _controller.checkCallPause(current);
    }

    public override void opWithPayload(byte[] seq, Ice.Current current)
    {
        _controller.checkCallPause(current);
    }

    public override void shutdown(Ice.Current current)
    {
        current.adapter.getCommunicator().shutdown();
    }

    internal BackgroundI(BackgroundControllerI controller)
    {
        _controller = controller;
    }

    private readonly BackgroundControllerI _controller;
}
